libindex/tempfile_unix.go

```diff
@@ -0,0 +1,30 @@
+//go:build unix && !linux
+
+package libindex
+
+import "os"
+
+type tempFile struct {
+	*os.File
+}
+
+func openTemp(dir string) (*tempFile, error) {
+	f, err := os.CreateTemp(dir, "*.fetch")
+	if err != nil {
+		return nil, err
+	}
+	return &tempFile{
+		File: f,
+	}, nil
+}
+
+func (t *tempFile) Reopen() (*os.File, error) {
+	return os.OpenFile(t.Name(), os.O_RDONLY, 0644)
+}
+
+func (t *tempFile) Close() error {
+	if err := os.Remove(t.Name()); err != nil {
+		return err
+	}
+	return t.File.Close()
+}
```

Comment:
```suggestion
func (t *tempFile) Close() error {
	return errors.Join(os.Remove(t.Name()), t.File.Close())
}
```

Should probably always remove and close, I think.

Reply:
makes sense to me